<launch>
    <arg name="wbr_model" />
    

    <!-- 将机器人模型加载到参数服务器当中 -->
    <param 
    name="robot_description" 
    command="$(find xacro)/xacro $(find wbr_pkg)/xacro/robot.xacro" />




    <!-- 启动 gazebo ，打开一个空世界（之后可以放一些障碍物我猜）-->
    <include file="$(find gazebo_ros)/launch/empty_world.launch" >
      <arg name = "paused" value = "true"/>
    </include>

    <!-- 在 gazebo 中显示机器人模型 -->
    <node pkg="gazebo_ros" type="spawn_model" name="model" args="
    -urdf 
    -model wbr_robot 
    -param robot_description"/>


    <!-- 以下部分是新加入的关于控制相关的内容 -->
      <!-- 1 Load joint controller configurations from YAML file to parameter server -->
        <rosparam file="$(find wbr_pkg)/yaml/robot_ctrl_fix.yaml" command="load"/>  
      <!-- 2 load the controllers -->
        <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
        output="screen" ns="/robot_wbr" args="
        joint_state_controller 
        Right_wheel_ctrl
        Left_wheel_ctrl
        "
        />
    <!-- Left_hip_ctrl
        Left_knee_ctrl
        Right_hip_ctrl
        Right_knee_ctrl -->

    
    
    <node
      name="joint_state_publisher"
      pkg="joint_state_publisher"
      type="joint_state_publisher" />
    <node
      name="robot_state_publisher"
      pkg="robot_state_publisher"
      type="robot_state_publisher" />

    <node pkg="wbr_pkg" type = "robot_node" name = "robot_node" output="screen"/>

</launch>


